### ISSUE TITLE
DatetimeIndex functionality: between_time

### ISSUE THREAD

**Issue by flipdazed** [vaexio/vaex#1226]
Thank you for reaching out and helping us improve Vaex!

**Description**
Can there be functionality provided to include: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.between_time.html

**Is your feature request related to a problem? Please describe.**
Select the rows of specific datetime range on any given day.

**Comment by JovanVeljanoski** [comment]
Hi, 

I guess this is mostly a convenience function, since you can already do this by manually specifying the filter. I am not sure if we want to implement this, as to not over extending the API. If you however tend to use this kind of thing a lot, and would appreciate the convenience, i think it would be quite easy to write your own vaex function/extension following [the examples here](https://vaex.readthedocs.io/en/latest/tutorial.html#Extending-Vaex)

What do you think @maartenbreddels ?

**Comment by maartenbreddels** [comment]
I agree with @JovanVeljanoski. I'd rather implement https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.between.html

so we can do:
`df = df[df.x.between(...)])`
and make `between` smart enough to recognize times and dates.
